draft-ietf-dnssd-privreq.xml

```diff
@@ -532,223 +543,171 @@ analysis can often reveal the service.
+    <t>In any protocol where the receiver of messages has to perform cryptographic
+    operations on those messages, there is a risk of a brute-force flooding
+    attack causing the receiver to expend excessive amounts of CPU time
+    (and battery power) just processing and discarding those messages.</t>
+  </section>
+
+  <section title="Resistance to Sender Impersonation">
+    <t>Sender impersonation is an attack wherein messages such as service offers are forged 
+    by entities who do not possess the corresponding secret key material. These attacks
+    may be used to learn the identity of a communicating party, actively or passively.</t>
+  </section>
+
+  <section title="Sender Deniability">
+    <t>Deniability of sender activity, e.g., of broadcasting a discovery request, may be 
+    desirable or necessary in some use cases. This property ensures that eavesdroppers
+    cannot prove senders issued a specific message destined for one or more peers. </t>
+  </section>
+</section>
+
+<section title="Operational Considerations">
+  <section title="Power Management">
+
+    <t>Many modern devices, especially battery-powered devices,
+    use power management techniques to conserve energy.
+    One such technique is for a device to transfer information about itself
+    to a proxy, which will act on behalf of the device for some functions,
+    while the device itself goes to sleep to reduce power consumption.
+    When the proxy determines that some action is required which only
+    the device itself can perform, the proxy may have some way
+    (such as Ethernet "Magic Packet") to wake the device.</t>
+
+    <t>In many cases, the device may not trust the network proxy
+    sufficiently to share all its confidential key material with the proxy.
+    This poses challenges for combining private discovery
+    that relies on per-query cryptographic operations,
+    with energy-saving techniques that rely on having (somewhat untrusted)
+    network proxies answer queries on behalf of sleeping devices.</t>
+
+  </section>
+
+  <section title="Protocol Efficiency">
+
+    <t>Creating a discovery protocol that has the desired security
+    properties may result in a design that is not efficient.
+    To perform the necessary operations the protocol may
+    need to send and receive a large number of network packets.
+    This may consume an unreasonable amount of network capacity
+    (particularly problematic when it's shared wireless spectrum),
+    cause an unnecessary level of power consumption
+    (particularly problematic on battery devices)
+    and may result in the discovery process being slow.</t>
+
+    <t>It is a difficult challenge to design a discovery protocol that has the
+    property of obscuring the details of what it is doing from unauthorized
+    observers, while also managing to do that quickly and efficiently.</t>
+
+  </section>
+
+  <section title="Secure Initialization and Trust Models">
+
```

Comment:
I wonder whether that really belongs in the discovery requirements. In fact, it is part of my doubt about the current pairing based solution: it solves the discovery, but also solves the much bigger problem of trusted interaction between clients and servers. You would expect that network architectures for unmanaged devices, p2p networks or IOT would already have a solution there. And if they had one, then discovery could just use whatever credentials are available for being part of the network.

Reply:
I'm fine with moving it out. It could go in an appendix, if we leave it in at all.